src/components/tx/ApprovalEditor/EditableApprovalItem.tsx

```diff
@@ -0,0 +1,66 @@
+import { Box, Button, IconButton, Stack, SvgIcon, Typography } from '@mui/material'
+import css from '@/components/tx/ApprovalEditor/styles.module.css'
+import type { ApprovalInfo } from './hooks/useApprovalInfos'
+
+import { ApprovalValueField } from './ApprovalValueField'
+import Track from '@/components/common/Track'
+import { MODALS_EVENTS } from '@/services/analytics'
+import { useFormContext } from 'react-hook-form'
+import { get } from 'lodash'
+import { EditOutlined } from '@mui/icons-material'
+import TokenIcon from '@/components/common/TokenIcon'
+import { shortenAddress } from '@/utils/formatters'
+import { useState } from 'react'
+
+const EditableApprovalItem = ({
+  approval,
+  name,
+  onSave,
+}: {
+  approval: ApprovalInfo
+  onSave: () => void
+  name: string
+}) => {
+  const { formState, setFocus } = useFormContext()
+
+  const { errors, dirtyFields } = formState
+
+  const fieldErrors = get(errors, name)
+  const isDirty = get(dirtyFields, name)
+
+  const [readOnly, setReadOnly] = useState(true)
+
+  const handleSave = () => {
+    onSave()
+    setReadOnly(true)
+  }
+
+  const handleEditMode = () => {
+    setReadOnly(false)
+    // We need to rerender such that select on focus triggers
+    setTimeout(() => setFocus(name), 0)
+  }
+
+  return (
+    <Stack direction="row" alignItems="center" gap={2} className={css.approvalField}>
+      <Box display="flex" flexDirection="row" alignItems="center" gap="4px">
+        <TokenIcon size={32} logoUri={approval.tokenInfo?.logoUri} tokenSymbol={approval.tokenInfo?.symbol} />
+        <Typography>{approval.tokenInfo?.symbol || shortenAddress(approval.tokenAddress)}</Typography>
```

Comment:
Imo this makes the alignment in the Box worse than before especially if we show a shortened address. Can it even happen that there is no symbol for a Token?
<img width="616" alt="Screenshot 2024-04-26 at 14 31 38" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/e666bf52-6e24-4bbe-907d-1ad6dfb4bf57">
<img width="429" alt="Screenshot 2024-04-26 at 14 31 53" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/9e735deb-13ad-4bca-8c82-1fd3ce5901b9">

Reply:
One alternative would be to show the Symbol as a Suffix to the value i.e. `Set allowance to 1 COW` but it seems hard to achieve in the input element. Maybe @TanyaEfremova has another idea?

Reply:
@usame-algan I think adding a suffix to the label w
<img width="699" alt="Screenshot 2024-04-26 at 10 17 42" src="https://github.com/safe-global/safe-wallet-web/assets/93707525/52406e7e-f13e-4500-9a50-904c1697dca0">
orks!

Reply:
I think we will have to convert these labels to template strings as it should be e.g. "Set WETH allowance to 5"
Or "Increase SAFE allowance by 20"